### DISCUSSION TITLE
How to monitor disk usage?

### DISCUSSION THREAD

**Discussion by cocowalla** [open-telemetry/opentelemetry-dotnet#4505] [Q&A]
OpenTelemetry is working great for process and runtime metrics, but I'd also like to report on remaining disk space, so I can configure alerts when it's running low.

Is there an instrumentation package available to do this?

**Comment by cijothomas** [comment]
> Is there an instrumentation package available to do this?

Nothing exists in OpenTelemetry repos for this today.

**Comment by cocowalla** [comment]
Are host/hardware metrics on the road map?

In the meantime, is it just a case of me creating my own `Gauge` to monitor disk usage, or is there another approach I should use? (I'm new to OpenTel, but liking what I'm seeing :)

**Reply by utpilla** [reply]
> Are host/hardware metrics on the road map?

Not anytime soon. Semantic Conventions has an [Experimental](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/document-status.md#:~:text=Equivalent%20to%20Experimental.-,Experimental,-Breaking%20changes%20are) spec for it: https://github.com/open-telemetry/semantic-conventions/blob/main/specification/metrics/semantic_conventions/hardware-metrics.md#hwphysical_disk--physical-disk-metrics. Things might change as they make progress towards stabilizing it.

> In the meantime, is it just a case of me creating my own Gauge to monitor disk usage, or is there another approach I should use
Yes, you could create your own instrument. Please keep in mind the [instrument selection criteria](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/supplementary-guidelines.md#instrument-selection:~:text=I%20want%20to%20measure,an%20Asynchronous%20UpDownCounter.) specified in the spec when choosing the right instrument for your requirement: `ObservableGauge` vs `ObservableUpDownCounter`.
